### PR TITLE
BTFS-1523: btfs rm should work for files with identical data blocks

### PR DESCRIPTION
**Pull request by taiyangc**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

  bugfix

* **What is the current behavior?** (You can also link to an open issue here)

  btfs rm does not work for files (small) with identical data blocks

* **What is the new behavior?** (You can also refer to a JIRA ticket here)

  fixed

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **What dependencies / modules need to be updated as pre-requisites?** (Include relevant PR links here)

* **Description of changes**

---

* **Please check if the PR fulfills these requirements**
- [x] The subject of this PR contains a JIRA ticket BTFS-xxxx (N/A for community PRs)
- [x] PR description and commit messages are [good and meaningful](https://chris.beams.io/posts/git-commit/)
- [ ] Unit tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **Please make sure the following procedures have been applied before requesting reviewers**
- [x] Code changes closely follow [Go Code Review Comments](https://github.com/golang/go/wiki/CodeReviewComments)
- [x] Code has been *rebased* against latest master (`git merge` not recommended, unless you know what you are doing)
- [x] Code changes have run through `go fmt`
- [x] Code changes have run through `go mod tidy`
- [x] All unit tests passed locally (`make test_go_test`)

### CODE REVIEW

File: core/commands/rm.go [hunk 1]
```diff
@@ -64,7 +64,7 @@ var RmCmd = &cmds.Command{
 			}
 
 			// Rm all child links
-			err = rmAllDags(req.Context, api, node, &results)
+			err = rmAllDags(req.Context, api, node, &results, map[string]bool{})
```
Comment (laipogo) [inline core/commands/rm.go]: `{}` is this a typo.

Comment (taiyangc) [reply core/commands/rm.go]: no, it's initializing an empty map

Comment (laipogo) [reply core/commands/rm.go]: cool

Comment (laipogo) [reply core/commands/rm.go]: i thought it's a function param.

**Pull request merged**